### DISCUSSION TITLE
Promisify Uploader.upload method

### DISCUSSION THREAD

**Discussion by vinodkhandelwal** [rpldy/react-uploady#196] [Q&A]
Hi,

How can i use Uploader.upload() method as a promise. I just wanted to await on this to completed before moving to next line.

```
await uploader.upload(); // Wait here till all uploads completes
history.push(ROUTES.HOME); // Move to Home page 

```

**Comment by yoavniran** [comment]
@vinodkhandelwal unfortunately, this isnt a behavior that can be supported by the Uploady uploader. 
The reason for this is that upload() doesnt necessarily initiate an upload directly. Batches can be added but not automatically uploaded (by setting `autoUpload` to false).

Only using events (ex: BATCH_START,  BATCH_FINISH, etc.) can you determine whether the uploads finished.

**Reply by vinodkhandelwal** [reply]
Thanks for the answer. yes that is what i am doing as of now. Keeping the number of files in state and comparing that with how many time BATCH_FINISH is getting called.
